\documentclass[aspectratio=43]{beamer}

\usepackage{amsmath, amsthm, amssymb}
\usepackage{empheq}
\usepackage{xcolor}
\newcommand{\R}{\mathbb{R}}
\usepackage{natbib}
\bibliographystyle{plain}


\title{Dimension reduction for single-cell and spatial RNA-seq using generalized bilinear models}
\author{\texorpdfstring{Phillip Nicol \\ Department of Biostatistics, Harvard University}{}}
\institute{NESS 2024 (Storrs, CT)}
\date{May 24, 2024}

\begin{document}

\titlepage

\begin{frame}{Introduction}

\begin{itemize}
\item Single-cell RNA-seq is a revolutionary technology that allows gene expression to be quantified at the level of individual cells. 

\item Data comes in the form of $I \times J$ count matrix $Y$. 

\item For large datasets $I \approx 10^4$ and $J \approx 10^7$. 

\item Dimension reduction is a critical first step before downstream analysis (clustering, visualization, etc.) 

\centering
\includegraphics[scale=0.6]{Fig/counts.png}

\end{itemize}
\end{frame}

\begin{frame}{PCA cannot be directly applied due to heterogeneous variances}
\centering 
\includegraphics[scale=0.4]{Fig/mean_variance_relationship.png}
\end{frame}

\begin{frame}{Standard approach is to transform the counts prior to PCA}

The counts are commonly pre-processed by computing the \textit{Pearson residual}:

\begin{equation*}
Z_{ij} := \frac{Y_{ij} - \hat{\mu}_{i}}{\sqrt{ \hat{\mu}_{i} - \hat{\alpha}_i \hat{\mu}_{i}^2}}
\end{equation*}

\begin{itemize}
\item scTransform (SCT) \cite{hafemeister2019normalization}: Estimate $\hat{\mu}$ and $\hat{\alpha}$ with a NB GLM.
\item APR \cite{lause2021analytic}: Fix $\hat{\alpha} = 0.01$ and use a closed-form approximation to $\hat{\mu}$. 
\end{itemize}
\end{frame}

\begin{frame}{PCA on Pearson residuals can fail to capture rare cell types}
If $\hat{\mu}_i$ is large then $Z_{ij} \approx Z_{ij'}$ even for very different counts.

\vspace{2em}

\centering
\includegraphics[scale=0.17]{Fig/single_marker_pca.png}
\end{frame}

\begin{frame}{PCA on Pearson residuals is sensitive to changes in baseline expression}

Starting with a dataset of technical replicates, scale each gene (row) by $\kappa_i \sim \text{Expo}(100)$:

\centering 
\includegraphics[scale=0.15]{Fig/ercc_scaled_pca.png}

\end{frame}



\begin{frame}{scGBM performs model-based dimension reduction}

\begin{empheq}[box=\fbox]{align*}
Y_{ij} &\sim \text{Pois}(\mu_{ij}) \\
\log(\mu_{ij}) &= \alpha_i + \beta_j + \sum_{m=1}^M \sigma_m u_{im} v_{jm} \\
\sigma_m &\sim \text{Expo}(a)
\end{empheq}

\centering
\includegraphics[scale=0.3]{Fig/gbm_model.png}
\raggedright

\vspace{-1em} 

$V \in \R^{J \times M}$ are the low-dimensional cell-embeddings
\end{frame}

\begin{frame}{Estimation with iteratively reweighted singular value decomposition}

\begin{itemize}
\item Can show that MLE is approximately equal to solving a weighted low rank problem.

\item {Define $\hat{X}^{(t)}$ to be the current estimate of $\hat{U} \hat{\Sigma} \hat{V}^{\top}$. The following update is used for the latent factors:

\begin{equation}
\hat{X}^{(t+1)} = \text{SVD}_{M, 1/a} \left( \hat{X}^{(t)} + \gamma( Y - \hat{\mu})     \right) 
\end{equation}
$\text{SVD}_{M, 1/a}(\cdot)$ computes the rank $M$ truncated SVD and then soft-thresholds the remaining singular values by $1/a$. }
\end{itemize}


\end{frame}

\begin{frame}{Faster estimation using scGBM-proj}

\begin{itemize}
\item When $J$ very large, first estimate $\hat{\alpha}$ and $\hat{U}$ using a smaller subset of cells. 
\item Then, holding these fixed, the parameters $\beta$ and $V \Sigma$ can be estimated by fitting $J$ GLMs in parallel. 
\item By analogy to PCA, we call this the \textit{projection method} (scGBM-proj)
\item 10X Brain Data (1000 genes, 1.3 million cells) fit in {\color{red} 55 minutes} compared to {\color{red} 6+ hours} for GLM-PCA \cite{townes2019feature}.
\end{itemize}

\end{frame}

\begin{frame}{scGBM outperforms GLM-PCA on simulated data}
\centering
\includegraphics[scale=0.35]{Fig/glmpca-simaccuracy.png}
\end{frame}


\begin{frame}{Comparing scGBM and GLM-PCA on real data}
\begin{itemize}
\item By thinning $Y$, can obtain independent $Y^*$, $Y^{**}$ and use one for validation.

\vspace{1em}

\end{itemize}
\centering
\includegraphics[scale=0.17]{Fig/oos_covid.png}
\end{frame}

\begin{frame}{Rare cell type simulation}
\centering
\includegraphics[scale=0.25]{Fig/single_marker_scGBM.png}
\end{frame}

\begin{frame}{Scaled data simulation}
Scaling rows only changes the baseline gene intercept $\alpha_i$.

\vspace{1em}

\centering 
\includegraphics[scale=0.15]{Fig/ercc_scaled_gbm.png}
\end{frame}

\begin{frame}{scGBM quantifies uncertainty in the low-dimensional embedding of cells}
\begin{itemize}
\item Quantify uncertainty in $V$ using classical method of inverting Fisher information matrix. 
\begin{itemize}
\item Full information matrix infeasible. Invert $M \times M$ submatrices holding other parameters fixed.
\item In simulation coverage only slightly below nominal level. 
\end{itemize}

\item Uncertainty could be useful in downstream applications. For example, consider $Y_{ij} \sim \text{Pois}(1)$ random noise clustered by leading software: 
\end{itemize}

\centering
\includegraphics[scale=0.14]{Fig/null_cluster_uncertainty.png}
\end{frame}

\begin{frame}{Cluster cohesion index measures cluster stability}
\centering
\includegraphics[scale=0.16]{Fig/cci.png}
\end{frame}

\begin{frame}{Extending to spatial transcriptomics}
\begin{itemize}
\item In spatial transcriptomics we also observe the spatial coordinate of cells $x_i \in \R^{d}$ ($d=2$ or $3$). 

\item {Can extend scGBM to identify factors that capture spatial features: 
\begin{equation*}
\text{argmin} -\ell(\alpha, \beta, U, \Sigma, V) + {\color{red} \mathcal{P}(V)}
\end{equation*}
where $\mathcal{P}(V)$ is a penalty that encourage nearby cells to have similar values for the embedding.}
\end{itemize}
\end{frame}

\begin{frame}{Edge-aware spatial smoothing}
\begin{itemize}
\item { 
Because boundaries between different spatial domain can be sharp, should not smooth uniformly in each direction:
\begin{equation*}
\mathcal{P}(V) = \sum_{j=1}^J \sum_{k \in N(j)} \omega_{jk} || V_{j \cdot} - V_{k \cdot} ||_2^2
\end{equation*}
where $\omega_{jk}$ can be set to $0$ if cells $j$ and $k$ are on opposite sides of boundary.

}
\end{itemize}


\centering
\includegraphics[scale=0.15]{Fig/edge_aware.png}
\end{frame}

\begin{frame}{Acknowledgements}
\begin{itemize}
\item Jeff Miller (Harvard Biostatistics)
\item Rafa Irizarry (DFCI Data Science)
\item NIH Cancer Training Grant (T32CA009337) 
\item Figures from article \cite{nicol2023model}.
\end{itemize}
\end{frame}

\begin{frame}{References}
\bibliography{sources}
\end{frame}

\end{document}